src/test.ts

```diff
@@ -14,85 +14,68 @@ function normalizeLineEndings(text: string): string {
   return text.replace(/\r\n?/g, "\n");
 }
 
-function compareToBaselines(baselineFolder: URL, outputFolder: URL) {
-  let baselineFiles: string[] = [];
+function getFiles(folder: URL) {
   try {
-    baselineFiles = fs.readdirSync(baselineFolder);
+    return fs.readdirSync(folder);
   } catch {
-    // do nothing
+    return [];
   }
+}
 
-  let outputFiles: string[] = [];
-  try {
-    outputFiles = fs.readdirSync(outputFolder);
-  } catch {
-    // do nothing
-  }
+function readFileContent(filePath: URL) {
+  return normalizeLineEndings(fs.readFileSync(filePath, "utf-8"));
+}
 
-  for (const file of new Set([...baselineFiles, ...outputFiles])) {
-    if (file.startsWith(".")) {
-      continue;
-    }
+function compareToBaselines(baselineFolder: URL, outputFolder: URL) {
+  const files = new Set([
+    ...getFiles(baselineFolder),
+    ...getFiles(outputFolder),
+  ]);
 
-    let baselineStats: fs.Stats | undefined;
-    try {
-      baselineStats = fs.statSync(new URL(file, baselineFolder));
-    } catch {
-      // do nothing
-    }
+  for (const file of files) {
+    if (file.startsWith(".")) continue;
 
-    let outputStats: fs.Stats | undefined;
-    try {
-      outputStats = fs.statSync(new URL(file, outputFolder));
-    } catch {
-      // do nothing
-    }
+    const baselinePath = new URL(file, baselineFolder);
+    const outputPath = new URL(file, outputFolder);
 
-    const baseline = baselineStats?.isFile()
-      ? normalizeLineEndings(
-          fs.readFileSync(new URL(file, baselineFolder)).toString(),
-        )
-      : null;
+    const isBaselineFile =
+      fs.existsSync(baselinePath) && fs.statSync(baselinePath).isFile();
```

Comment:
It's actually advised to *not* call `exists` before other calls, because think can potentially be racy and the file can go away after that call.

https://nodejs.org/docs/latest/api/fs.html#fsexistspath-callback

>Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile(), or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.